frontend/src/views/CellGuide/components/common/OntologyDagView/index.tsx

```diff
@@ -112,13 +109,13 @@ export default function OntologyDagView({
   const [width, setWidth] = useState(inputWidth);
   const [height, setHeight] = useState(inputHeight);
 
-  const [initialTransformMatrix, setInitialTransformMatrix] = useState<
-    typeof initialTransformMatrixDefault
-  >(initialTransformMatrixDefault);
+  const zoomRef = useRef<(ProvidedZoom<SVGSVGElement> & ZoomState) | null>(
```

Comment:
We will use zoomRef to keep track of the current zoom object, which lets us manipulate the transform matrix.